# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [43]:
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

##### 1  We will run a two sample significance test comparing callback proportions for the two races. Since we have over 2000 entries for each sample, our sample size is large enough for the CLT to hold. 

In [14]:
print(len(data[data.race == 'b']), len(data[data.race == 'w']))

2435 2435


##### 2  
       Null Hypothesis: There is no significant difference in callback rates based on applicant race (assumes Pw-Pb = 0 or Pw=Pb = P)

       Hypothesis: The callback rates vary based on applicant race

##### 3 Margin of error, Confidence Inrerval (We'll use 95%, z-score = 1.96), and P-value

In [34]:
print('w proportions: total = ',len(data[data.race == 'w']), ' call = ', len(data[data.race == 'w'][data.call == 1]), ' no call = ', len(data[data.race == 'w'][data.call == 0]))
print('b proportions: total = ',len(data[data.race == 'b']), ' call = ', len(data[data.race == 'b'][data.call == 1]), ' no call = ', len(data[data.race == 'b'][data.call == 0]))

w proportions: total =  2435  call =  235  no call =  2200
b proportions: total =  2435  call =  157  no call =  2278


C:\Users\sashkabear\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  if __name__ == '__main__':
C:\Users\sashkabear\Anaconda3\lib\site-packages\ipykernel\__main__.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  from ipykernel import kernelapp as app


In [54]:
a_tot, a_1, a_0 = 2435, 235, 2200
b_tot, b_1, b_0 = 2435, 157, 2278 
Pw, Pb = 235/2435, 157/2435 #Measures Callback Proportion For each race
Pdiff = Pw-Pb #Measured Difference in Proportion
P_diff_null = 0 #Assumed difference in proportions with the null Hypothesis
P = (a_1 + b_1)/(a_tot + b_tot) #Best estimate of consistent population proportion (i.e. given the null hypothesis)b
SD_null = math.sqrt(2*P*(1-P)/a_tot) #SD of sampling distribution estimate
SD_measured = math.sqrt(Pw*(1-Pw)/2435 + Pb*(1-Pb)/2435) #Measured SD of proportion difference
d = SD_measured * 1.96
z_score = (Pdiff - P_diff_null)/SD_null 
print(Pw, Pb, Pdiff, P, SD_null, SD_measured, d , z_score)

0.09650924024640657 0.06447638603696099 0.032032854209445585 0.08049281314168377 0.007796894036170457 0.0077833705866767544 0.015255406349886438 4.108412152434346


95% CI: Confident that there is a 95% Chance that the true difference in callback proportion is within d = 0.015 of Sample proportion of 0.032. 95% CI= 0.016 - 0.047

Margin of error: d= 0.015

P-value: Our z-score = 4.1, from z-table z~4 results in probability of about 0.99997,  so p-value < 0.00003. We can reject the Null Hypothesis

##### 4 Story
Based on sample data we estblished a 95% CI for the range of values for difference in callback proportions based on race that could be attributed to chance. This range gives us a reasonable basis for judgement of our measured z score. If the z-score of the difference in our samples is above 1.96, we can expect this to be a chance occurence in less than 5% of the cases, in other words - not likely. Since our measured z-score ~ 4.1, we can expect this result to be a chance occurence in <.003% of the cases, in other words - Extremely unlikely. Thus we can reject the null hypothesis in favor of the alternate hypothesis - that there is in fact a difference in interview call back rates with respect to applicant's race.

##### 5
While we did determine that the difference in callback proportions based on race is very unlikely to occur by chance, and that we can assume race does affect interview callback rates, we can not make a judjement on whether race in the most important factor without conducting additional analysis. From the data above we could check for variables such as: years of experience, age, number of jobs applied, etc, and see how they affect call back rates. It is also possible that certain important factors were not considered in the presented experiment - number of applicants per job, method of resume submission, number of contacts at the company, etc. While we can conclude that percieved race affects callback race, there could be a number of factors that influence employer's decision to a greater extent.
